quickwit/quickwit-search/src/root.rs

```diff
@@ -204,6 +205,14 @@ pub(crate) fn validate_request(
         )));
     }
 
+    if search_request.query_ast.len() > searcher_context.query_string_max_length {
```

Comment:
I'm not fond of enforcing that on the query_ast. On the other hand, it might be difficult to do something at the user query level. Maybe we should have a limit not in number of chars but in number of terms?

Reply:
Ok.
The query equivalent that was easiest to handle was `search_request.query_ast`

I have a few questions about this response.
- Is it necessary to change `query_ast` from a json format to a structure again for simplicity?
- Which is the recognition that indicates which is the `term` here?
  (Is it in the `QueryAst` field??)